pennylane/ops/op_math/controlled_class.py

```diff
@@ -98,6 +112,20 @@ class Controlled(SymbolicOp):
 
     """
 
+    # pylint: disable=no-self-argument
+    @operation.classproperty
+    def __signature__(cls):  # pragma: no cover
+        # this method is defined so inspect.signature returns __init__ signature
+        # instead of __new__ signature
+        # See PEP 362
```

Comment:
nice!

Reply:
black magic required more black magic, but at least it should be working now.